### PR TITLE
String/exercise 06

### PR DESCRIPTION
**Pull request by ntnghia0320**

### CODE REVIEW

File: src/main/java/StringExercise06.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+public class StringExercise06 {
+    public String shortcutString(String str) {
+        int n = str.length() - 1;
```
Comment (voquanghoa) [inline src/main/java/StringExercise06.java]:
```suggestion

```

File: src/main/java/StringExercise06.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        for (int i = 0; i < n; i++) {
+            if (str.charAt(i) == str.charAt(i + 1)) {
+                count++;
+            } else if (count > 1) {
+                result.append(str.charAt(i));
+                result.append(count);
+                count = 1;
+            } else {
+                result.append(str.charAt(i));
+            }
```
Comment (voquanghoa) [inline src/main/java/StringExercise06.java]:
```suggestion
            } else {
                   result.append(str.charAt(i));
                   if (count > 1) result.append(count);
            }
```

File: src/main/java/StringExercise06.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        }
+
+        if (count > 1) {
+            result.append(str.charAt(n - 1));
+            result.append(count);
+        } else {
+            result.append(str.charAt(n));
+        }
+
+        return String.valueOf(result);
```
Comment (voquanghoa) [inline src/main/java/StringExercise06.java]:
```suggestion
        return result.toString();
```

File: src/main/java/StringExercise06.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            } else if (count > 1) {
+                result.append(str.charAt(i));
+                result.append(count);
+                count = 1;
+            } else {
+                result.append(str.charAt(i));
+            }
+        }
+
+        if (count > 1) {
```
Comment (voquanghoa) [inline src/main/java/StringExercise06.java]: Same above

File: src/main/java/StringExercise06.java (hunk 1 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise06.java]: Done

File: src/main/java/StringExercise06.java (hunk 2 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise06.java]: Done

File: src/main/java/StringExercise06.java (hunk 3 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise06.java]: Done

File: src/main/java/StringExercise06.java (hunk 4 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise06.java]: Done

**Pull request merged**